packages/gallery/src/components/gallery/proGallery/slideshowView.js

```diff
@@ -436,11 +434,23 @@ class SlideshowView extends React.Component {
     }
   };
 
+  getFirstIdx(itemIdx) {
+    const activeItem = this.props.galleryStructure.galleryItems[itemIdx];
+    const activeItemId = activeItem.itemId;
+    const firstItemInstance = this.props.galleryStructure.galleryItems.find((item) => item.itemId === activeItemId);
+    return firstItemInstance.idx;
+  }
+
   scrollToThumbnail(itemIdx, scrollDuration) {
     //not to confuse with this.props.actions.scrollToItem. this is used to replace it only for thumbnail items
 
     this.props.actions.eventsListener(GALLERY_CONSTS.events.THUMBNAIL_CLICKED, this.props);
-    this.scrollToIndex({ itemIdx, scrollDuration });
+    const activeItemFirstIdx = this.getFirstIdx(this.state.activeIndex);
+    const firstTargetItemIdx = this.getFirstIdx(itemIdx);
+    const distance = firstTargetItemIdx - activeItemFirstIdx;
+    const targetIdx = this.state.activeIndex + distance;
+
+    this.scrollToIndex({ itemIdx: targetIdx, scrollDuration });
```

Comment:
I think this is more elegant - 

```suggestion
const activeIdx = this.state.activeIndex;
const targetIdx = activeIdx + (this.getFirstIdx(itemIdx) - this.getFirstIdx(activeIdx));
this.scrollToIndex({ itemIdx: targetIdx, scrollDuration });
```